### PR TITLE
Reply to inline comments

### PR DESCRIPTION
**Pull request by wandersoncferreira**
@charignon I had put to myself another requirement for working with PR reviews from emacs: be able to respond to comments inline. I believe this is mentioned #9 

This PR adds such functionality with a few caveats:

- Any new comment (other than in top level portion of buffer) that is *immediately* preceded by a previous comment will be treated as a reply
- I could only find a Github API that accepts single reply calls, so it may be slow or get rate limited if you have *X* number of replies. (the value of X is unknown to me)
- This only works if `github-review-view-comments-in-code-lines` is enabled.
- I tried to change as minimal as possible the current data structures used in the code, but I am afraid I cannot feature flag *everything* without a lot of duplication, if you see a way forward I am all ears :)

Give it a try 👍

### CODE REVIEW

**Review by charignon — CHANGES_REQUESTED** [review]
Left a few comments inline to resolve, thanks again for all your work!

**Comment by charignon** [comment]
Additionally, feel free to squash the two empty commits from the stack by rebasing on top of the current master

**Comment by wandersoncferreira** [comment]
Thanks @charignon  for your feedbacks!
BTW: All my comments were made from Github Review Diff buffer 🙃  🙃

**Comment by wandersoncferreira** [comment]
IMO this is good to go, the main change is guarded on feature flag and the other changes in the comment data structure do not introduce much overhead for users.

File: github-review.el [hunk 1]
```diff
@@ -175,6 +180,39 @@ CALLBACK will be called back when done"
                :errorback #'github-review-errback
                :callback callback)))
 
+(defun github-review-post-review-replies (pr-alist replies callback)
+  "Submit replies to review comments inline."
+  (deferred:$
+    (deferred:loop
```
Comment (charignon) [inline github-review.el]: Is it the same as deferred parallel? Could you double check that these happen indeed in parallel?

**Review by charignon — APPROVED** [review]
Approved, left one comment inline for you to address / test before merging @wandersoncferreira

File: github-review.el (hunk 1 above)
Comment (wandersoncferreira) [reply github-review.el]: Sequential :/ I changed the code to use `deferred:parallel` now. 👍

File: github-review.el [hunk 2]
```diff
@@ -195,7 +225,7 @@ CALLBACK will be called back when done"
 
 (defun github-review-previous-comment? (l)
   "Return t if L, a string, is a comment from previous review."
-  (string-prefix-p "~ " l))
+  (string-prefix-p "~" l))
```
[the remark this answers is not in the document]
Comment (wandersoncferreira) [reply github-review.el]: made days later

**Pull request merged**